src/data/ab-attrs/reflect-stat-stage-change-ab-attr.ts

```diff
@@ -0,0 +1,46 @@
+import type { Pokemon } from "#app/field/pokemon";
+import { globalScene } from "#app/global-scene";
+import { StatStageChangePhase } from "#app/phases/stat-stage-change-phase";
+import type { BooleanHolder } from "#app/utils";
+import type { BattleStat } from "#enums/stat";
+import { AbAttr } from "./ab-attr";
+
+/**
+ * Attribute to reflect stat-lowering effects from moves and abilities
+ * back to their source.
+ * Used for {@link https://bulbapedia.bulbagarden.net/wiki/Mirror_Armor_(Ability) | Mirror Armor}.
+ * @extends AbAttr
+ */
+export class ReflectStatStageChangeAbAttr extends AbAttr {
+  /**
+   * If the current {@linkcode StatStageChangePhase} will lower the ability
+   * source's stat(s), queue a new phase targeting the effect source
+   * with the same stat stage changes.
+   * @param pokemon the {@linkcode Pokemon} with this ability
+   * @param simulated if `true`, suppresses changes to game state
+   * @param source the {@linkcode Pokemon} applying the stat change
```

Comment:
```suggestion
   * @param source the {@linkcode Pokemon} applying the original stat change
```
makes things a bit clearer maybe?